test/serviceprofiles/serviceprofiles_test.go

```diff
@@ -0,0 +1,170 @@
+package serviceprofiles
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/linkerd/linkerd2/testutil"
+)
+
+var TestHelper *testutil.TestHelper
+
+type testCase struct {
+	args           []string
+	deployName     string
+	expectedRoutes []string
+	namespace      string
+	sourceName     string
+	spName         string
+}
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(m.Run())
+}
+
+func TestServiceProfiles(t *testing.T) {
+
+	testNamespace := TestHelper.GetTestNamespace("serviceprofile-test")
+	out, _, err := TestHelper.LinkerdRun("inject", "testdata/tap_application.yaml")
+	if err != nil {
+		t.Fatalf("linkerd inject command failed\n%s", out)
+	}
+
+	out, err = TestHelper.KubectlApply(out, testNamespace)
+	if err != nil {
+		t.Fatalf("kubectl apply command failed\n%s", out)
+	}
+
+	// wait for deployments to start
+	for _, deploy := range []string{"t1", "t2", "t3", "gateway"} {
+		if err := TestHelper.CheckPods(testNamespace, deploy, 1); err != nil {
+			t.Error(err)
+		}
+
+		if err := TestHelper.CheckDeployment(testNamespace, deploy, 1); err != nil {
+			t.Error(fmt.Errorf("Error validating deployment [%s]:\n%s", deploy, err))
+		}
+	}
+
+	testCases := []testCase{
+		{
+			sourceName:     "tap",
+			namespace:      testNamespace,
+			deployName:     "deploy/t1",
```

Comment:
Any reasons why we aren't testing `deploy/t2` and `deploy/t3`?

Reply:
The focus for these tests was for different ways to generate service profiles. So we want to make sure we have one deploy that generates tap data and one deploy that is capable of using a swagger file (deploy/t3, is an HTTP service, so we can add a mock swagger file for it).

Also `deploy/t2` is pretty much similar to `deploy/t1` except it always fails.